R/init.R

```diff
@@ -231,6 +231,7 @@ init <- function(data,
       if (!is.null(landing_module)) {
         do.call(landing_module$server, c(list(id = "landing_module_shiny_id"), landing_module$server_args))
       }
+      teal_lockfile_tracker(process)
```

Comment:
can we group the related functionalities close together? 

```suggestion
      teal_lockfile_tracker(teal_lockfile_invoke())
```

Alternatively we can have one "main" function
```
      create_lockfile()
```

Reply:
I actually think that introducing yet another level (`create_lockfile`) would be nice to keep file-based-approach separate to the future-based-approach

pseudo-code
```
create_lockfile <- function() {
  if ( file condition ) {
    teal_lockfile_read()
  } else { 
    process <- teal_lockfile_process_invoke()
    teal_lockfile_process_tracker(process)
  }
}
```

Then we can isolate all file based logic from the current `teal_lockfile_invoke` - conceptually there is no need to _invoke_ if file is provided. This would also simplify the current `teal_lockfile_tracker` because you don't have to handle empty `process` argument.

Reply:
This makes a lot of sense